state/execution.go

```diff
@@ -312,15 +319,36 @@ func (blockExec *BlockExecutor) Commit(
 	)
 
 	// Update mempool.
-	err = blockExec.mempool.Update(
+	go blockExec.asyncUpdateMempool(unlockMempool, block, state.Copy(), deliverTxResponses)
+
+	return res.Data, res.RetainHeight, err
+}
+
+// updates the mempool with the latest state asynchronously.
+func (blockExec *BlockExecutor) asyncUpdateMempool(
+	unlockMempool func(),
+	block *types.Block,
+	state State,
+	deliverTxResponses []*abci.ResponseDeliverTx,
```

Comment:
@ValarDragon function signature changes here from 

`abciResponse *abci.FinalizeBlockResponse`

to 

`deliverTxResponses []*abci.ResponseDeliverTx`

because of the backport to v0.37.x, I'm not sure how big an issue this is please ack here :bow: